VisualStudio.gitignore

```diff
@@ -1,16 +1,18 @@
-## Ignore Visual Studio temporary files, build results, and
-## files generated by popular Visual Studio add-ons.
-
-# User-specific files
+# temporary files generated by Visual Studio specific to the local user
+# contains profile information and other metadata
+# remove: if no longer targeting Visual Studio 2013 or earlier
 *.suo
 *.user
 *.userosscache
 *.sln.docstates
-
-# User-specific files (MonoDevelop/Xamarin Studio)
+#
+# new location for Visual Studio 2015
+.vs/
+#
+# specific to MonoDevelop and Xamarin Studio
 *.userprefs
 
-# Build results
+# build output folders generated by Visual Studio
```

Comment:
This is the simplest structure I could come up with, and can be described as (my regex-fu is rusty, apologies in advance):

```
section: 
  context+ 
  rule+

context: comment+;
comment:  '#' ~[\r\n]* CRLF;
rule: ~[\r\n]* CRLF

CRLF
 : '\r'? '\n' 
 | '\r'
 ;
```

TL;DR:
 - each section should have at least one line of context
 - you can have multiple rules in a section